oracle/src/events/processAMBCollectedSignatures/index.js

```diff
@@ -50,6 +56,48 @@ function processCollectedSignaturesBuilder(config) {
 
         logger.info(`Processing CollectedSignatures ${colSignature.transactionHash}`)
         const message = await homeBridge.methods.message(messageHash).call()
+        const parsedMessage = parseAMBMessage(message)
+
+        if (ORACLE_HOME_TO_FOREIGN_ALLOWANCE_LIST || ORACLE_HOME_TO_FOREIGN_BLOCK_LIST) {
+          const sender = parsedMessage.sender.toLowerCase()
+          const executor = parsedMessage.executor.toLowerCase()
+
+          if (ORACLE_HOME_TO_FOREIGN_ALLOWANCE_LIST) {
+            const allowanceList = await readAccessListFile(ORACLE_HOME_TO_FOREIGN_ALLOWANCE_LIST, logger)
+            if (!allowanceList.includes(executor)) {
+              if (ORACLE_HOME_TO_FOREIGN_CHECK_SENDER === 'true') {
+                if (!allowanceList.includes(sender)) {
+                  logger.info(
+                    { sender, executor },
+                    'Validator skips a message. Neither sender nor executor addresses are in the allowance list.'
+                  )
+                  return
+                }
+              } else {
+                logger.info({ executor }, 'Validator skips a message. Executor address is not in the allowance list.')
+                return
+              }
+            }
+          } else if (ORACLE_HOME_TO_FOREIGN_BLOCK_LIST) {
+            const blockList = await readAccessListFile(ORACLE_HOME_TO_FOREIGN_BLOCK_LIST, logger)
+            if (blockList.includes(executor)) {
+              logger.info({ executor }, 'Validator skips a message. Executor address is in the block list.')
+              return
+            }
+            if (ORACLE_HOME_TO_FOREIGN_CHECK_SENDER === 'true' && blockList.includes(sender)) {
+              logger.info({ sender }, 'Validator skips a message. Sender address is in the block list.')
+              return
+            }
+          }
+        }
+
+        if (parsedMessage.dataType & 128) {
```

Comment:
Does it make sense to wrap it by a function to have a meaningful name like 'isLaneManagedByExtension'? I would also recommend to add a comment to not forget why this bit is being checked.

Reply:
Done in 5437d63